lib/util/exec/docker/index.ts

```diff
@@ -198,13 +198,19 @@ export async function generateDockerCommand(
 
   result.push(...prepareVolumes([localDir, cacheDir, ...volumes]));
 
+  const customEnvVariables = config.customEnvVariables || {};
   if (envVars) {
     result.push(
       ...uniq(envVars)
-        .filter((x) => typeof x === 'string')
+        .filter(
+          (x) => typeof x === 'string' && customEnvVariables[x] === undefined
+        )
         .map((e) => `-e ${e}`)
     );
   }
+  Object.entries(customEnvVariables).forEach(([key, val]) => {
+    result.push(`-e ${key}=${val}`);
```

Comment:
I have a doubt here. If these custom env variables are for regular child process as well as Docker sidecar containers, then shouldn't they already be set as desired in env in code in `lib/util/exec/index.ts`? i.e. meaning only `-e` is necessary instead of the full value?

Reply:
I guess it makes sense but for my understanding can you point me to the place where we set `process.env.KEY=VALUE` (assuming it's set somewhere)?

Reply:
Actually, if I revert changes in this file, the test fails:

```
    -   "docker run --rm --name=renovate_image --label=renovate_child -v \"/tmp/renovate/github/some/repo/\":\"/tmp/renovate/github/some/repo/\" -v \"/tmp/renovate/cache/\":\"/tmp/renovate/cache/\" -e CUSTOM_KEY=CUSTOM_VALUE -w \"/tmp/renovate/github/some/repo/\" renovate/image bash -l -c \"echo hello\"",
    +   "docker run --rm --name=renovate_image --label=renovate_child -v \"/tmp/renovate/github/some/repo/\":\"/tmp/renovate/github/some/repo/\" -v \"/tmp/renovate/cache/\":\"/tmp/renovate/cache/\" -w \"/tmp/renovate/github/some/repo/\" renovate/image bash -l -c \"echo hello\"",
```

I was expecting to see `-e CUSTOM_KEY`. I guess I am doing something wrong in `lib/util/exec/index.ts`. 🤔

Reply:
If you're not seeing it then it would imply it's not part of `envVars`. It's constructed here: https://github.com/PicnicSupermarket/renovate/blob/a707b4619b2f91cca7630aae7c0d5d5675a9c839/lib/util/exec/index.ts#L134 so perhaps we need to add the customEnvVariables to those?

Reply:
You're right to be sceptical.  Added one more commit, now the implementation is much simpler.